examples/call/contracts/Universal.sol

```diff
@@ -22,46 +22,56 @@ contract Hello is UniversalContract {
         gateway = GatewayZEVM(gatewayAddress);
     }
 
-    function onCall(
-        MessageContext calldata context,
-        address zrc20,
-        uint256 amount,
-        bytes calldata message
-    ) external override onlyGateway {
-        string memory name = abi.decode(message, (string));
-        emit HelloEvent("Hello on ZetaChain", name);
-    }
-
-    function onRevert(
-        RevertContext calldata revertContext
-    ) external override onlyGateway {
-        emit RevertEvent("Revert on ZetaChain", revertContext);
-    }
-
     function call(
         bytes memory receiver,
         address zrc20,
         bytes calldata message,
-        uint256 gasLimit,
+        CallOptions memory callOptions,
         RevertOptions memory revertOptions
     ) external {
-        (, uint256 gasFee) = IZRC20(zrc20).withdrawGasFeeWithGasLimit(gasLimit);
+        (, uint256 gasFee) = IZRC20(zrc20).withdrawGasFeeWithGasLimit(
+            callOptions.gasLimit
+        );
         if (!IZRC20(zrc20).transferFrom(msg.sender, address(this), gasFee))
             revert TransferFailed();
         IZRC20(zrc20).approve(address(gateway), gasFee);
-        gateway.call(receiver, zrc20, message, gasLimit, revertOptions);
+        gateway.call(receiver, zrc20, message, callOptions, revertOptions);
+    }
+
+    function withdraw(
+        bytes memory receiver,
+        uint256 amount,
+        address zrc20,
+        RevertOptions memory revertOptions
+    ) external {
+        (address gasZRC20, uint256 gasFee) = IZRC20(zrc20).withdrawGasFee();
+        uint256 target = zrc20 == gasZRC20 ? amount + gasFee : amount;
+        if (!IZRC20(zrc20).transferFrom(msg.sender, address(this), target))
+            revert TransferFailed();
+        IZRC20(zrc20).approve(address(gateway), target);
+        if (zrc20 != gasZRC20) {
+            if (
```

Comment:
in the previus transfer you use if() { revert } and in this one if() revert. Let's use the same in all. I prefer with {} to make it more readeable

Reply:
Do you use curly braces even for one-liner `if`s?

Reply:
A:

```solidity
if (gas == 0) revert InvalidGasLimit();
```

or B:

```solidity
if (gas == 0) revert InvalidGasLimit();
```

Reply:
83ab10b